docs/observability/ovn-observability.md

```diff
@@ -0,0 +1,122 @@
+# Observability
+
+## Introduction
+
+Observability feature uses OVN sampling functionality to generate samples with requested metadata when
+specific OVS flows are matched. To see the generated samples, a binary called `ovnkube-observ` is used.
+This binary allows printing the samples to stdout or writing them to a file.
+
+Currently, supports observability for:
+- Network Policy
+- (Baseline) Admin Network Policy
+- Egress firewall
+- UDN isolation
+- Multicast ACLs
+
+More features are planned to be added in the future. 
+
+## Motivation
+
+Networking observability is an important feature to verify the expected networking behavior in a cluster and
+to debug existing problems.
+Ovn-kubernetes makes use of many abstraction layers (through NBDB, logical flows, openflow flows and datapath flows) 
+that translate kubernetes feature into very specific rules that apply 
+to each packet in the network. Therefore, even though there are ways to see what OVS/OVN is doing with a particular packet, 
+there is no way to know why.
+
+We aim to solve this problem by providing a way for ovn-kubernetes to generate packet samples enriched with metadata 
+that can be easily correlated back to kubernetes objects or other human-readable pieces of information that provide 
+insights of what ovn-kubernetes is doing with a packet and why.
+
+### User-Stories/Use-Cases
+
+- As a user I want to make sure that the network policies/egress firewalls/etc. are correctly enforced in my cluster.
+- As a cluster admin I want to check why some traffic is allowed or dropped.
+
+## How to enable this feature on an OVN-Kubernetes cluster?
+
+To enable this feature, use `--observability` flag with `kind.sh` script or `--enable-observability` flag with `ovnkube` binary.
+
+To see the samples, use `ovnkube-observ` binary, use `-h` to see allowed flags.
+
+This feature requires OVS 3.4 and linux kernel 6.11.
+
+As of Aug 2024, the kernel and OVS 3.4 need to be built from the source, therefore to try this feature you need to:
+- rebuild the kernel with the current master branch from [Linus' tree](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git)
+  - to rebuild on fedora: https://docs.fedoraproject.org/en-US/quick-docs/kernel-build-custom/#_building_a_vanilla_upstream_kernel
+- Build an ovn-kubernetes image that uses the latest OVS/OVN code:
+`OVS_BRANCH=main make -C dist/images fedora-dev`
+- Start kind with that image, use `-ov localhost/ovn-kube-fedora-dev:latest` flag with `kind.sh` script.
```

Comment:
Unfortunately things have changed since the last review and now Fedora:rawhide has OVS 3.4 builds available.  When #4658 merges we'll get OSV 3.4 for free.  Let's change this paragraph to only mention the kernel.